fuel-tx/src/checked_transaction.rs

```diff
@@ -39,12 +42,18 @@ impl core::fmt::Display for Checks {
 /// of `Checked` outside the `fuel-tx` crate.
 ///
 /// The inner data is immutable to prevent modification to invalidate the checking.
+/// A single exception to this is provided so that extension traits can set the flags
+/// after they have checked some properties. For instance fuel-vm is required to check
+/// the predicates of a transaction. This function is hidden from the docs, and must
+/// not be used for anything else.
 ///
 /// If you need to modify an inner state, you need to get inner values
 /// (via the `Into<(Tx, Tx ::Metadata)>` trait), modify them and check again.
 ///
-/// # Dev note: Avoid serde serialization of this type. Since checked tx would need to be
-/// re-validated on deserialization anyways, it's cleaner to redo the tx check.
+/// # Dev note: Avoid serde serialization of this type.
+///
+/// Since checked tx would need to be re-validated on deserialization anyways,
+/// it's cleaner to redo the tx check.
 #[derive(Debug, Clone, Eq, PartialEq, Hash)]
 pub struct Checked<Tx: IntoChecked> {
```

Comment:
Maybe it makes sense to move `Checked` into `fuel-vm`, because it is where we really start to use it=)  In this case you can pass `Interpreter` that would check predicates=)

Reply:
Maybe that should be a separate PR after this one? I'd also like to find a way to force tx checking, maybe by encoding that into the type itself.

Reply:
We have to move lots of stuff to over, because transcation builder and `checked` mod of each transaction type depend on this as well.

Reply:
PR for this: https://github.com/FuelLabs/fuel-vm/pull/310